lib/analysis/modules/ies_enkf.c

```diff
@@ -0,0 +1,1059 @@
+*  for each subsequent iteration we store the current mask in module_data->obs_mask */
+  if (!module_data->obs_mask0)
+     module_data->obs_mask0 = bool_vector_alloc_copy( obs_mask );
+
+  if (module_data->obs_mask)
+    bool_vector_free( module_data->obs_mask );
+
+  module_data->obs_mask = bool_vector_alloc_copy( obs_mask );
+}
+
+
+
+/*
+ * -------------------------------------------------------------------------------------------------------------
+ * I E n K S
+ * IEnKS update (IES searching the solution in ensemble subspace)
+ * -------------------------------------------------------------------------------------------------------------
+*/
+
+void ies_enkf_updateA( void * module_data,
+                       matrix_type * A ,      // Updated ensemble A retured to ERT.
+                       matrix_type * Yin ,    // Ensemble of predicted measurements
+                       matrix_type * Rin ,    // Measurement error covariance matrix (not used)
+                       matrix_type * dObs ,   // Actual observations (not used)
+                       matrix_type * Ein ,    // Ensemble of observation perturbations
+                       matrix_type * Din ,    // (d+E-Y) Ensemble of perturbed observations - Y
+                       const module_info_type * module_info,
+                       rng_type * rng) {
+
+   ies_enkf_data_type * data = ies_enkf_data_safe_cast( module_data );
+
+   int nrobs_msk     =bool_vector_size( data->obs_mask ); // Total number of observations
+   int nrobs_inp     =matrix_get_rows( Yin );             // Number of active observations input in current iteration
+   int nrobs         =nrobs_inp;                          // Number of selected active observations
+
+
+   int ens_size_msk  =bool_vector_size(data->ens_mask);   // Total number of realizations
+   int ens_size      =matrix_get_columns( Yin );          // Number of active realizations in current iteration
+
+   int state_size    = matrix_get_rows( A );
+
+   double rcond;
+   int nrsing=0;
+
+   if (data->state_size == 0) data->state_size=state_size;
+
+
+   data->iteration_nr         = data->iteration_nr+1 ;
+   data->ies_steplength       = ies_enkf_config_get_ies_steplength( data->config );
+   data->ies_subspace         = ies_enkf_config_get_ies_subspace( data->config );
+   data->ies_inversion        = ies_enkf_config_get_ies_inversion( data->config );
+   data->ies_logfile          = ies_enkf_config_get_ies_logfile( data->config );
+   data->ies_debug            = ies_enkf_config_get_ies_debug( data->config );
+
+   data->truncation           = ies_enkf_config_get_truncation( data->config );
+   data->subspace_dimension   = ies_enkf_config_get_enkf_subspace_dimension( data->config );
+
+   data->gauss_newton_conv    = ies_enkf_config_get_gauss_newton_conv( data->config );
+
+   if (data->iteration_nr > 3) data->ies_steplength=data->ies_steplength/2;
+   if (data->iteration_nr > 6) data->ies_steplength=data->ies_steplength/4;
+
+
+   FILE *fp;
```

Comment:
I have not seen the `freopen()` function before - but it seems to "steal" the `stdout` stream in a highly non-local way; I don't like it :-(

I would prefer if you instead:
```
data->log_fp = fopen(data->ies_logfile, "w/a");
..
fprintf(data->log_fp, ".....", ...);
...
fclose(data->log_fp);
```